README.md

```diff
@@ -1 +1,196 @@
+
+```YAML
+# Required to indicate this is a `metrics.yaml` file
+$schema: moz://mozilla.org/schemas/glean/metrics/1-0-0
+
+crash:
+  exception:
+    type: event
+    description: |
+      Event to record crashes caused by unhandled exceptions
+    notification_emails:
+      - crashes@example.com
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+    expires:
+      2099-01-01
+    send_in_pings:
+      - crash
+    extra_keys:
+      cause:
+        description: The cause of the crash
+      message:
+        description: The exception message
+```
+
+By way of explanation, this creates a metric called `exception` within a metric category called `crash`.  There is a brief description and the required notification, bug, data review, and expiry fields.  Then we have the `send_in_pings` field with a value of `- crash`.  This means that we will send the crash data via a custom ping named `crash` (which we haven’t created, yet).  Finally we have the `extra_keys` field with two keys defined, `cause` and `message`.  This will allow us to send a couple of pieces of additional information along with the event.
+
+### Add Custom Ping
+
+Now we need to define the custom ping by creating a pings.yaml file in the same directory as we just created the `metrics.yaml` file.  We already know what the name of the ping is, `crash`, so the `pings.yaml` file should look like this:
+
+```YAML
+# Required to indicate this is a `pings.yaml` file
+$schema: moz://mozilla.org/schemas/glean/pings/1-0-0
+
+crash:
+  description: >
+    A ping to transport crash data
+  include_client_id: true
+  notification_emails:
+    - crash@example.com
+  bugs:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+  data_reviews:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+```
+
+Before we can use our newly defined metric or ping, we need to build the application.  This will cause the [glean_parser](https://github.com/mozilla/glean_parser/) to work its magic and generate the API files that represent the metrics we defined.
+
+**NOTE:** Just a little advice from my personal experience here, but I would recommend doing a Clean on your project before building _any_ time you have modified one of the Glean YAML files.  Sometimes it doesn't regenerate the files without the Clean so just something to keep in mind if you aren’t seeing your changes to the YAML files show up in the project.
+
+We now have a couple of tasks to perform back in the MainActivity.kt file in order to make use of the metric and ping we defined.  First, we need to add an import line:
+
+```Kotlin
+import org.mozilla.gleancrashexample.GleanMetrics.Pings
+```
+
+And then we need to register our custom ping, so right after `Glean.initialize(this)` add the following line:
```

Comment:
Is the "right after" part true? I don't think that's strictly required, so I'd leave it out before it becomes tribal knowledge :D

Reply:
Gotcha